src/main/java/org/apache/maven/shared/dependency/analyzer/asm/ConstantPoolParser.java

```diff
@@ -168,20 +174,46 @@ static Set<String> parseConstantPoolClassReferences( ByteBuffer buf )
                     break;
             }
         }
+        
         Set<String> result = new HashSet<>();
-        for ( Integer aClass : classes )
+        
+        for ( Integer classRef : classReferences )
         {
-            String className = stringConstants.get( aClass );
-
-            // filter out things from unnamed package, probably a false-positive
-            if ( isImportableClass( className ) )
+            addClassToResult( result, stringConstants.get( classRef ) );
+        }
+        
+        for ( Integer typeRef : typeReferences )
+        {
+            String typeName = stringConstants.get( typeRef );
+            
+            Type type = Type.getType( typeName );
+            switch ( type.getSort() )
```

Comment:
This is strange. Why is a switch used with only a single case? An if block seems more straightforward.

Reply:
Btw. the module doesn't build with Java 19. I have to switch to an older Java (like 11) before the build so it goes through.

Reply:
The repo, yes. I'll file an issue.

Reply:
https://issues.apache.org/jira/browse/MSHARED-1205